src/common/ios/iOSPlatform.ts

```diff
@@ -1,140 +1,161 @@
+                this.generateSuccessPatterns(),
+            () =>
+                Q(IOSPlatform.RUN_IOS_FAILURE_PATTERNS)).process(runIosSpawn);
+    }
+
+    public enableJSDebuggingMode(): Q.Promise<void> {
+        // Configure the app for debugging
+        if (this.targetType === IOSPlatform.deviceString) {
+            // Note that currently we cannot automatically switch the device into debug mode.
+            Log.logMessage("Application is running on a device, please shake device and select 'Debug in Chrome' to enable debugging.");
+            return Q.resolve<void>(void 0);
+        }
+
+        const iosDebugModeManager = new IOSDebugModeManager(this.iosProjectPath);
+
+        // Wait until the configuration file exists, and check to see if debugging is enabled
+        return Q.all<boolean | string>([
+            iosDebugModeManager.getSimulatorRemoteDebuggingSetting(),
+            this.getBundleId(),
+        ])
+            .spread((debugModeEnabled: boolean, bundleId: string) => {
+                if (debugModeEnabled) {
+                    return Q.resolve(void 0);
+                }
+
+                // Debugging must still be enabled
+                // We enable debugging by writing to a plist file that backs a NSUserDefaults object,
+                // but that file is written to by the app on occasion. To avoid races, we shut the app
+                // down before writing to the file.
+                const childProcess = new ChildProcess();
+
+                return childProcess.execToString("xcrun simctl spawn booted launchctl list")
+                    .then((output: string) => {
+                        // Try to find an entry that looks like UIKitApplication:com.example.myApp[0x4f37]
+                        const regex = new RegExp(`(\\S+${bundleId}\\S+)`);
+                        const match = regex.exec(output);
+
+                        // If we don't find a match, the app must not be running and so we do not need to close it
+                        return match ? childProcess.exec(`xcrun simctl spawn booted launchctl stop ${match[1]}`) : null;
+                    })
+                    .then(() => {
+                        // Write to the settings file while the app is not running to avoid races
+                        return iosDebugModeManager.setSimulatorRemoteDebuggingSetting(/*enable=*/ true);
+                    })
+                    .then(() => {
+                        // Relaunch the app
+                        return this.runApp();
+                    });
+            });
+    }
+
+    public prewarmBundleCache(): Q.Promise<void> {
+        return this.remoteExtension.prewarmBundleCache(this.platformName);
+    }
+
+    public getRunArgument(): string[] {
+        let runArguments: string[] = [];
+        if (this.targetType) {
+            runArguments.push(`--${this.targetType}`);
+        }
+
+        if (this.target) {
+            runArguments.push(this.target);
```

Comment:
Latest RN throws an error in case if device name is missing:
```
rn run-ios --device
Scanning 578 folders for symlinks in /Users/kotikov.vladimir/sandbox/baz/node_modules (20ms)
Found Xcode project baz.xcodeproj
Could not find device with the name: "true".
Choose one of the following:
```

This might be a bug in RN or an intended behavior, in either case it might make sense to either warn user about this and stop operation or try enumerating available devices (using `ios-deploy --detect`) and choose the first one, for example.

@iTOYS what do you think?